server/controllers/download-by-content-id.js

```diff
@@ -99,6 +101,24 @@ module.exports = exports = async (req, res, next) => {
 		await dl.appendAll();
 	}
 	else {
+		log.count('article-download-start')
+
+		dl.on('error', (err) => {
+			log.error('DOWNLOAD_ARTICLE_ERROR', {
+				event: 'DOWNLOAD_ARTICLE_ERROR',
+				error: err.stack || err
+			});
+			log.count('article-download-error');
+			res.status(500).end();
+		});
```

Comment:
**Suggestion:** As this is the same as the `dl.on('error'...)` event in the `if` statement, we can move this above the if/else statement only needing to write it once.

```javascript
	log.count('archive-download-start');
	
	dl.on('error', (err) => {
		log.error('DOWNLOAD_ARTICLE_ERROR', {
			event: 'DOWNLOAD_ARTICLE_ERROR',
			error: err.stack || err
		});
		log.count('article-download-error');
		res.status(500).end();
	});
	
	if (dl.downloadAsArchive) {
		...
	} else {
		...
	}
```

Reply:
There is a difference in the two callbacks, this suggestion would not work for our need. Resolving this conversation.